tutorials/ExploringDeutschJozsaAlgorithm/DeutschJozsaAlgorithmTutorial.ipynb

```diff
@@ -620,7 +620,83 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "# Part V. What's Next?\n",
+    "# Part V.  Different function types - what if our function is neither balanced nor constant?\n",
+    "\n",
+    "The setup for the Deutsch-Josza algorithm presupposes that the function we're trying to identify is perfectly constant or perfectly balanced in its output (indeed, we haven't even considered the possibility that an input outside of this space could be used). Although the algorithm will not converge to a solution in the same manner that we've seen so far when selecting another type of function as input, we can still derive some intuition for how quantum states behave probabilistically by considering some outside-the-box cases.\n",
+    "\n",
+    "Before thinking about particular cases, it's worthwhile to consider how to classify (in a purely qualitative fashion) our function $f(x)$ with respect to its outputs being balanced or constant. One might consider $f(x)$ to be a superposition of balanced and constant functions, such that it conveys incomplete properties of both types. $f(x)$ could then be said to be balanced for some subset of its inputs and constant for the remainder. We could ask ourselves, \"How balanced is $f(x)$?\" in the same way that we might ask, for a qubit, \"How 'close' is the qubit to a $|0\\rangle$ or $|1\\rangle$ state?\"\n",
+    "\n",
+    "For example, consider the following oracle that introduces an unbalanced (read: 'constant') aspect into what you might recognize as the operation for `PhaseOracle_Xmod2`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "operation PhaseOracle_AlmostBalanced (x : Qubit[]) : Unit {\n",
+    "     // Supply a -1 phase to the last qubit's |1> basis state\n",
+    "     Z(x[Length(x) - 1]);\n",
+    "     \n",
+    "     // Subject to a tight set of controls, supply an additional -1 phase to a limited set of states as follows:\n",
+    "     // -Define the controls for a Controlled Z gate as all but the top two qubits\n",
+    "     // -Apply the phase to the qubit immediately below the MSB\n",
+    "     // -The net result, using an example of 8 qubits, will affect two states:\n",
+    "     //    |01111111>\n",
+    "     //    |11111111>\n",
```

Comment:
I'm not sure I follow the argument here: we take a balanced function and flip the phases of two states, but these two states used to have different phases (since they have different lowest bit, the first Z gate will only introduce the -1 phase to 11111111), so in the end the negative and positive phases will remain balanced? 

It might make sense to use the same `Controlled Z(x[0..Length(x) - 2], x[Length(x) - 1]);` as in the last example here as well, since that one is guaranteed to switch the phase of just one state.

Reply:
Agreed, there was a discrepancy with what I pushed. Using the controlled-Z operation you mentioned gives an accurate picture. I'll update the video and the commentary accordingly.

Reply:
Resolved in the latest push - using `x[0..Length(x) - 2]` as the control and the last qubit as the target.